src/index.js

```diff
@@ -26,13 +27,28 @@ module.exports = function () {
                 return;
             }
 
-            var code = [
-                    `import shimWorker from 'rollup-plugin-bundle-worker';`,
-                    `export default new shimWorker(${JSON.stringify(paths.get(id))}, function (window, document) {`,
-                    `var self = this;`,
-                    fs.readFileSync(id, 'utf-8'),
-                    `\n});`
-                ].join('\n');
+            var code = `
+                import shimWorker from 'rollup-plugin-bundle-worker';
+                const workerFunction = function(window, document) {
+                  var self = this;
+                  ${fs.readFileSync(id, 'utf-8')}
+                };`;
+
+            const newShimWorker = `new shimWorker(${JSON.stringify(paths.get(id))}, workerFunction);`;
+
+            if (userOptions.lazy) {
+                code += `
+                let WorkerObject;
```

Comment:
We now cache the workerObject too, that way we don't call `workerFunction.toString` on recreation, which could be a performance drain.